src/mlpack/bindings/python/print_input_processing.hpp

```diff
@@ -122,21 +122,48 @@ void PrintInputProcessing(
   std::cout << prefix << "# Detect if the parameter was passed; set if so."
       << std::endl;
   if (!d.required)
-  {
-    std::cout << prefix << "if " << d.name << " is not None:" << std::endl;
+  { 
+    if (GetArmaType<T>() == "row" || GetArmaType<T>() == "col")
+    {
+      std::cout << prefix << "if " << d.name << " is not None:" << std::endl;
```

Comment:
You could move this little bit outside the `if` since it appears in both the body and the body of the `else`. :)